examples/problems/co2ptflashproblem.hh

```diff
@@ -498,6 +498,24 @@ private:
             fs.setViscosity(FluidSystem::gasPhaseIdx, FluidSystem::viscosity(fs, paramCache, FluidSystem::gasPhaseIdx));
         }
 
+        // determine the component total fractions
+        // TODO: duplicated code here, while should be refactored out when we swithing
+        // to starting from total mole fractions
+        Dune::FieldVector<Scalar, numComponents> z(0.0);
+        Scalar sumMoles = 0.0;
+        for (unsigned phaseIdx = 0; phaseIdx < numPhases; ++phaseIdx) {
+            for (unsigned compIdx = 0; compIdx < numComponents; ++compIdx) {
+                Scalar tmp = Opm::getValue(fs.molarity(phaseIdx, compIdx) * fs.saturation(phaseIdx));
+                z[compIdx] += Opm::max(tmp, 1e-8);
+                sumMoles += tmp;
```

Comment:
If `tmp < 1.0e-8` then `sumMoles` is, strictly speaking, not the sum of the elements of `z`.  Is that okay?

Reply:
It is originally through the commit OPM/opm-common@1d2db38172539e0dd722e884e647824db06ceacb 

It is used multiple places and it might be used to treat unexpected values.  We might not need it but I do not think it will affect the actual results. 

But I will make a follow up PR to check whether we can get rid of all of them (there are four usages, and three of them are in opm-common). 

It is possible that it was introduced in the early stage of the development to get things run through.

Reply:
I see your point now. Let me think a little.

Reply:
I changed the code to be as follows, 

```c++
                Scalar tmp = getValue(fs.molarity(phaseIdx, compIdx)) * saturation;
                tmp = max(tmp, 1.e-8);
                z[compIdx] += tmp;
                sumMoles += tmp;
```